### ISSUE TITLE
Multiple asserts in failing test case yields wrong total number of tests in junit report

### ISSUE THREAD

**Issue by jpdutoit** [supermarin/xcpretty#206]
If a test has multiple asserts that fail, it causes the `testcase` element in the junit report to be duplicated, once for each assert failure. This causes the total test count to vary based on which tests fail.

For example, in the following snippet the `testcase` element for the test `testExample2` appears twice. 
```
<testcase classname="TestClass" name="testExample2">
<failure message="(object == nil) failed: "<NSString: 0x7a6b3a60>"">
Tests/TestClass.m:10
</failure>
</testcase>
<testcase classname="TestClass" name="testExample2">
<failure message="(object == nil) failed: "<NSString: 0x7a6b3a60>"">
Tests/TestClass.m:15
</failure>
</testcase>
```

**Comment by nacuteodor** [comment]
Hey @jpdutoit , 
I 've written a tool that parse the TestSummaries plist file from Logs/Test folder and generates the JUnit report xml file: [https://github.com/nacuteodor/ProcessTestSummaries ProcessTestSummaries] .
You shouldn't have this problem with that tool. 
Hope that will help you.

**Comment by nacuteodor** [comment]
Hey @jpdutoit , 
I 've written a tool that parse the TestSummaries plist file from Logs/Test folder and generates the JUnit report xml file: [https://github.com/nacuteodor/ProcessTestSummaries ProcessTestSummaries] .
You shouldn't have this problem with that tool. 
Hope that will help you.

**Comment by hejhj** [comment]
@nacuteodor Thank you. It works.

**Comment by dlednik** [comment]
+1

I have same issue too. Can this be fixed soon pls.

**Comment by QuintinWillison** [comment]
We're seeing this issue too. It would seem, to me, that the JUnit output that xcpretty is providing is incorrect. Surely there should only ever be a single instance of testcase for each classname? Seems like a pretty big oversight to me. Disappointing.

**Comment by aditjain** [comment]
@dlednik @QuintinWillison did you this one resolved?

**Comment by dlednik** [comment]
@aditjain no. I just set my XCode to stop testing on 1st error it encounters in a test. So that gives me the correct number of failures now :)

**Comment by QuintinWillison** [comment]
@aditjain also, no here. I ended up writing my own post-processor to 'conform' the output (Jenkins setting). Currently internal company code but we might consider open sourcing.

**Comment by aditjain** [comment]
Thanks @dlednik and @QuintinWillison for the response. 

@dlednik you mean continueAfterFailure to false?

**Comment by dlednik** [comment]
@aditjain yes

**Comment by aditjain** [comment]
@dlednik doesnot work for me! :( 

@QuintinWillison I understand its company's internal, but general advise to proceed will be highly appreciated.

**Comment by dlednik** [comment]
@aditjain Try this

```Swift
class XCTestCommon: XCTestCase {   
    override func setUp() {
        super.setUp()
        self.continueAfterFailure = false
    }
    
    override func tearDown() {
        super.tearDown()
    }
}
```

Make all your test classes inherit this one instead of XCTestCase.
It will enable this for every test run ;)

**Comment by aditjain** [comment]
@dlednik  thats exactly what I have! :(

**Comment by dlednik** [comment]
@aditjain then it should work. It does for me.

This is how your test classes should look:  

```Swift
class MyTestClass: XCTestCommon {
    func testFails() {
        XCTFail("Fail1")
        XCTFail("Fail2")
        XCTFail("Fail3")
        XCTFail("Fail4")
    }
}
```

**Comment by aditjain** [comment]
@dlednik i think XCTFail might not be viable solution since i will not always know where my test fails.

**Comment by dlednik** [comment]
It only demonstrates that if the parameter is taken into the account it will only report first fail in XCode and not all four of them

**Comment by revathysivakumar** [comment]
Any update on this bug?
